### PR TITLE
Enable basic travis test suites

### PR DESCRIPTION
**Pull request by stvstnfrd**

### CODE REVIEW

**Comment by stvstnfrd** [comment]
@jmbowman @mzulqarnain1 Sorry for the overlap. @nasthagiri had asked me to ensure we were actually running tests, so I've added travis-ci integration to perform linting and some sanity checks that ensure we can actually render an instance of the XBlock.

Much of the underlying infrastructure was in place; it just hadn't been called :(

**Comment by stvstnfrd** [comment]
Commit history is messy; I'll squash and cleanup prior to merge.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - 3.5
+env:
+  - TOX_ENV=django111
+  - TOX_ENV=django20
+  - TOX_ENV=django21
+  - TOX_ENV=django22
+matrix:
+  include:
+    - python: 3.5
+      env: TOX_ENV=csslint
```
Comment (jmbowman) [inline .travis.yml:17]: These could just go into the list in `env:` above, couldn't they?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      env: TOX_ENV=pylint
+before_install:
+  - "pip install -U pip"
+  - export BOTO_CONFIG=/dev/null
+cache:
+  directories:
+    - $HOME/.cache/pip
+install:
+  - "make requirements"
+  - "pip install coveralls"
```
Comment (jmbowman) [inline .travis.yml:32]: We've switched to codecov in most places; had a lot of problems with coveralls reliability a while back.

File: sql_grader/tests/test_display.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+"""
+Test basic XBlock display function
+"""
+from __future__ import absolute_import
```
Comment (jmbowman) [inline sql_grader/tests/test_display.py:5]: You shouldn't need this import now that we've dropped Python 2.7 support.

File: requirements/test.in [hunk 4]
```diff
@@ -0,0 +1,3 @@
+coverage
+edx-opaque-keys
+mock
```
Comment (jmbowman) [inline requirements/test.in:3]: Maybe just use `unittest.mock` instead now that we're on Python 3?

File: tox.ini [hunk 5]
```diff
@@ -2,13 +2,17 @@
 envlist =
     csslint
     eslint
-    py36
+    py{36}-django{111,20,21,22}
```
Comment (jmbowman) [inline tox.ini:5]: Should probably test with py35 instead, as that's what we currently use in production.

File: tox.ini [hunk 6]
```diff
@@ -2,13 +2,17 @@
...
     pycodestyle
     pylint
 
 [testenv]
 deps =
-    coverage
+    -rrequirements/test.in
```
Comment (jmbowman) [inline tox.ini:11]: We should add proper OEP-18 support later in a separate PR.

File: .travis.yml (hunk 2 above)
Comment (stvstnfrd) [reply .travis.yml:28]: Ah, I cribbed much of this from the ORA2 xblock, which still uses coveralls.

Can we punt for now?

Regardless, good to know for the future.

File: tox.ini (hunk 6 above)
Comment (stvstnfrd) [reply tox.ini:11]: Yeah, apologies; this was a bit quick-and-dirty to get some feedback in the form of tests.

File: .travis.yml (hunk 1 above)
Comment (stvstnfrd) [reply .travis.yml]: Done. Note: I had shamelessly cargo-culted this from `edx-ora2`

File: sql_grader/tests/test_display.py (hunk 3 above)
Comment (stvstnfrd) [reply sql_grader/tests/test_display.py]: Done.

File: requirements/test.in (hunk 4 above)
Comment (stvstnfrd) [reply requirements/test.in]: Done.

File: tox.ini (hunk 5 above)
Comment (stvstnfrd) [reply tox.ini]: Done.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+sudo: false
+python:
+  - 3.5
+env:
+  - TOX_ENV=csslint
+  - TOX_ENV=django111
+  - TOX_ENV=django20
+  - TOX_ENV=django21
+  - TOX_ENV=django22
+  - TOX_ENV=django22
```
Comment (mzulqarnain1) [inline .travis.yml:15]: Following 2 lines are redundant.

File: sql_grader/tests/test_display.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        Ensure we can render a basic XBlock
+        """
+        student_view = self.xblock.student_view()
+        html = student_view.content
+        self.assertIsNotNone(html)
+        self.assertNotEqual('', html)
+        self.assertIn('sql_grader_block', html)
+
+
+if __name__ == '__main__':
```
Comment (mzulqarnain1) [inline sql_grader/tests/test_display.py:53]: We can remove this.

File: .travis.yml (hunk 7 above)
Comment (stvstnfrd) [reply .travis.yml]: Done.

File: sql_grader/tests/test_display.py (hunk 8 above)
Comment (stvstnfrd) [reply sql_grader/tests/test_display.py]: Done.

File: .travis.yml (hunk 2 above)
Comment (stvstnfrd) [reply .travis.yml]: Done.

**Pull request merged**